lib/src/main/java/graphql/nadel/instrumentation/parameters/NadelInstrumentationOnErrorParameters.kt

```diff
@@ -0,0 +1,25 @@
+package graphql.nadel.instrumentation.parameters
+
+import graphql.execution.instrumentation.InstrumentationState
+
+data class NadelInstrumentationOnErrorParameters(
+    val message: String,
+    val exception: Throwable,
+    val errorType: ErrorType,
+    val errorData: Any?,
+    private val instrumentationState: InstrumentationState?,
+) {
+    fun <T : InstrumentationState?> getInstrumentationState(): T? {
+        @Suppress("UNCHECKED_CAST") // trust the caller
+        return instrumentationState as T?
+    }
+}
+
+data class OnServiceExecutionErrorData(
```

Comment:
Can we do

```kotlin
sealed interface NadelOnErrorData {
    data class OnServiceExecution(
        val executionId: String,
        val serviceName: String,
    ) : NadelOnErrorData
}
```

And change

```diff
-     val errorData: Any?,
+     val errorData: NadelOnErrorData,
```

Just so it's stricter and easier to consume. I hate seeing `Any?` in API and then having to guess what the possible values are.

Reply:
that's awesome!